packages/core/src/parseAmplifyOutputs.ts

```diff
@@ -0,0 +1,363 @@
+		url,
+		api_key,
+		model_introspection,
+	} = amplifyOutputsDataProperties;
+
+	const GraphQL: APIGraphQLConfig = {
+		endpoint: url,
+		defaultAuthMode: getGraphQLAuthMode(default_authorization_type),
+		region: aws_region,
+		apiKey: api_key,
+		modelIntrospection: model_introspection,
+	};
+
+	return {
+		GraphQL,
+	};
+}
+
+function parseNotifications(
+	amplifyOutputsNotificationsProperties?: AmplifyOutputsNotificationsProperties,
+): NotificationsConfig | undefined {
+	if (!amplifyOutputsNotificationsProperties) {
+		return undefined;
+	}
+
+	const { aws_region, channels, pinpoint_app_id } =
+		amplifyOutputsNotificationsProperties;
+
+	if (
+		!channels.includes('IN_APP_MESSAGING') &&
+		!channels.includes('APNS') &&
+		!channels.includes('FCM')
+	) {
+		return undefined;
+	}
+
+	let InAppMessaging: InAppMessagingConfig | undefined;
+	let PushNotification: PushNotificationConfig | undefined;
+
+	if (channels.includes('IN_APP_MESSAGING')) {
+		InAppMessaging = {
+			Pinpoint: {
+				appId: pinpoint_app_id,
+				region: aws_region,
+			},
+		};
+	}
+
+	if (channels.includes('APNS') || channels.includes('FCM')) {
+		PushNotification = {
+			Pinpoint: {
+				appId: pinpoint_app_id,
+				region: aws_region,
+			},
+		};
+	}
+
+	if (PushNotification && InAppMessaging) {
+		return {
+			PushNotification,
+			InAppMessaging,
+		};
+	}
+
+	if (InAppMessaging) {
+		return {
+			InAppMessaging,
+		};
+	}
+
+	if (PushNotification) {
+		return {
+			PushNotification,
+		};
+	}
+}
+
+export function parseAmplifyOutputs(
+	amplifyOutputs: AmplifyOutputs,
+): ResourcesConfig {
+	const resourcesConfig: ResourcesConfig = {};
+
+	if (amplifyOutputs.storage) {
+		resourcesConfig.Storage = parseStorage(amplifyOutputs.storage);
+	}
+
+	if (amplifyOutputs.auth) {
+		resourcesConfig.Auth = parseAuth(amplifyOutputs.auth);
+	}
+
+	if (amplifyOutputs.analytics) {
+		resourcesConfig.Analytics = parseAnalytics(amplifyOutputs.analytics);
+	}
+
+	if (amplifyOutputs.geo) {
+		resourcesConfig.Geo = parseGeo(amplifyOutputs.geo);
+	}
+
+	if (amplifyOutputs.data) {
+		resourcesConfig.API = parseData(amplifyOutputs.data);
+	}
+
+	if (amplifyOutputs.notifications) {
+		resourcesConfig.Notifications = parseNotifications(
+			amplifyOutputs.notifications,
+		);
+	}
+
+	return resourcesConfig;
+}
+
+const authModeNames: Record<AuthType, GraphQLAuthMode> = {
+	AMAZON_COGNITO_USER_POOLS: 'userPool',
+	API_KEY: 'apiKey',
+	AWS_IAM: 'iam',
+	AWS_LAMBDA: 'lambda',
+	OPENID_CONNECT: 'oidc',
+};
+
+function getGraphQLAuthMode(authType: AuthType): GraphQLAuthMode {
+	return authModeNames[authType];
+}
+
+const providerNames: Record<
+	AmplifyOutputsOAuthIdentityProvider,
+	OAuthProvider
+> = {
+	GOOGLE: 'Google',
+	LOGIN_WITH_AMAZON: 'Amazon',
+	FACEBOOK: 'Facebook',
+	SIGN_IN_WITH_APPLE: 'Apple',
+};
+
+function getOAuthProviders(
+	providers?: AmplifyOutputsOAuthIdentityProvider[],
+): OAuthProvider[] {
+	return (providers ?? []).map(provider => providerNames[provider]);
```

Comment:
Nit: Just use a default value instead
```suggestion
	providers: AmplifyOutputsOAuthIdentityProvider[] = [],
): OAuthProvider[] {
	return providers.map(provider => providerNames[provider]);
}
```